integration_tests/index/test_search.py

```diff
@@ -1181,6 +1183,19 @@ def test_csv_structure(clirunner, pseudo_ls8_type, ls5_telem_type,
     assert lines[0] == _EXPECTED_OUTPUT_HEADER
 
 
+def test_query_dataset_multi_product(index: Index, ls5_dataset_w_children: Dataset):
+    # We have one ls5 level1 and its child nbar
+    dc = Datacube(index)
+
+    # Can we query a single product name?
+    datasets = dc.find_datasets(product='ls5_nbar_scene')
+    assert len(datasets) == 1
+
+    # Can we query multiple products?
+    datasets = dc.find_datasets(product=['ls5_nbar_scene', 'ls5_level1_scene'])
```

Comment:
Users often assume that `product=("a", "b")` should work the same as `product=["a", "b"]`, would be nice to test that case as well, except it will break right now, see discussion in main section.

Reply:
Sorry, I thought I had pushed a commit. Pushing now.